rabbit-escape-ui-swing/src/rabbitescape/ui/swing/SwingGameLaunch.java

```diff
@@ -381,6 +387,7 @@ public int addToken( int tileX, int tileY, Token.Type ability )
         {
             graphics.playSound( "place_token" );
         }
+        solutionRecorder.append( new PlaceTokenAction( tileX, tileY ) );
         return now;
```

Comment:
I followed where the mouse clicks went, and inserted this into the business end. I guess it's because ability selection is a mouse button click and that is handled differently to a token drop where you need to find out where on the main game panel a click has happened and map it to a location in the game's world.